containers/haskell/.devcontainer/Dockerfile

```diff
@@ -1,25 +1,50 @@
-# Use the [Choice] comment to indicate option arguments that should appear in VS Code UX
-#
-# [Choice] Haskell version: 9, 8
-ARG VARIANT=8
-FROM haskell:${VARIANT}
-
-# Use the [Option] comment to specify true/false arguments that should appear in VS Code UX
-#
+FROM debian:bullseye-slim
+
+ENV LANG C.UTF-8
+
 # [Option] Install zsh
 ARG INSTALL_ZSH="true"
 # [Option] Upgrade OS packages to their latest versions
 ARG UPGRADE_PACKAGES="false"
 
-# Install needed packages and setup non-root user. Use a separate RUN statement to add your own dependencies
+# Install needed packages and setup non-root user.
+# Use a separate RUN statement to add your own dependencies
 ARG USERNAME=vscode
 ARG USER_UID=1000
 ARG USER_GID=$USER_UID
 COPY library-scripts/*.sh /tmp/library-scripts/
 RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
     && /bin/bash /tmp/library-scripts/common-debian.sh "${INSTALL_ZSH}" "${USERNAME}" "${USER_UID}" "${USER_GID}" "${UPGRADE_PACKAGES}" "true" "true"\
-    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* /tmp/library-scripts
+    && rm -rf /tmp/library-scripts
+
+# common haskell + stack dependencies
+RUN apt-get install -y --no-install-recommends \
+      dpkg-dev \
+      gcc \
+      libc6-dev \
+      libffi-dev \
+      libgmp-dev \
+      libnuma-dev \
+      libtinfo-dev \
+      zlib1g-dev \
+    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
```

Comment:
I think the other definitions combines two layers like these into one.

Reply:
Fixed!